### PR TITLE
Photo Library Enhancement.

### PR DESCRIPTION
**Pull request by jamessa**
iOS 9+ is required.

the flow becomes,

* When you take a photo from Camera, it will be saved to Camera Roll and send to Unity.
* When you select a photo from Library, it will only send to Unity.

Permission is checked right before the action. Just in time permission.

### CODE REVIEW

**Comment by jamessa** [comment]
fixes issue #10 #11

File: Assets/Plugins/iOS/SPECameraViewController.mm [hunk 1]
```diff
@@ -424,30 +448,39 @@ - (void)setupCamera {
     }
 }
 
-- (void)cameraDenied {
+- (void)requestForPhotoLibraryAuthorization {
+    if ([PHPhotoLibrary authorizationStatus] != PHAuthorizationStatusAuthorized) {
+        [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
+            [self displayPrivacySettingsInstruction];
```
Comment (kokuyoku82) [inline Assets/Plugins/iOS/SPECameraViewController.mm]: check not equal to `PHAuthorizationStatusAuthorized`

File: Assets/Plugins/iOS/SPECameraViewController.mm [hunk 2]
```diff
@@ -355,6 +363,22 @@ - (void)shutterAction:(UIButton *)sender {
                 image = [image fixOrientation];
                 __block NSString *imagePath = [self saveImage:image];
                 
+                [self requestForPhotoLibraryAuthorization];
```
Comment (kokuyoku82) [inline Assets/Plugins/iOS/SPECameraViewController.mm]: The method must be added completion block, subsequent events add to this block .

File: Assets/Plugins/iOS/SPECameraViewController.mm [hunk 3]
```diff
@@ -299,11 +304,14 @@ - (void)helpAction:(UIButton *)sender {
...
 - (void)closeAction:(UIButton *)sender {
     [self dismissViewControllerAnimated:YES completion:^{
-
+        
     }];
 }
 
 - (void)albumAction:(UIButton *)sender {
+    // Request for Photo Library read and write permission
+    [self requestForPhotoLibraryAuthorization];
```
Comment (kokuyoku82) [inline Assets/Plugins/iOS/SPECameraViewController.mm]: The method must be added completion block, subsequent events add to this block.

**Pull request merged**